packages/dcos-launch/extra/launch/aws.py

```diff
@@ -119,18 +118,35 @@ def get_stack(self, info):
         except Exception as ex:
             raise launch.util.LauncherError('StackNotFound', None) from ex
 
-    def test(self, info, test_cmd):
-        launch.util.check_testable(info)
-        details = self.describe(info)
-        test_host = details['masters'][0]['public_ip']
-        with ssh.tunnel.tunnel(info['ssh_user'], info['ssh_private_key'], test_host) as test_tunnel:
-            return test_util.runner.integration_test(
-                tunnel=test_tunnel,
-                dcos_dns=test_host,
-                master_list=[m['private_ip'] for m in details['masters']],
-                agent_list=[a['private_ip'] for a in details['private_agents']],
-                public_agent_list=[a['private_ip'] for a in details['public_agents']],
-                aws_access_key_id=info['aws_access_key_id'],
-                aws_secret_access_key=info['aws_secret_access_key'],
-                region=info['aws_region'],
-                test_cmd=test_cmd)
+    def describe(self, info):
+        cf = self.get_stack(info)
+        return {
+            'masters': launch.util.convert_host_list(cf.get_master_ips()),
+            'private_agents': launch.util.convert_host_list(cf.get_private_agent_ips()),
+            'public_agents': launch.util.convert_host_list(cf.get_public_agent_ips())}
+
+
+class BareClusterLauncher(DcosCloudformationLauncher):
+    """ Launches a homogenous cluster of plain AMIs intended for onprem DC/OS
+    """
+    def create(self, config):
+        """ Amend the config to add a template_body and the appropriate parameters
+        """
+        config.update({
+            'template_body': test_util.aws.template_by_instance_type(config['instance_type']),
+            'template_parameters': yaml.dump({
+                'KeyName': config['aws_key_name'],
+                'AllowAccessFrom': config['admin_location'],
+                'ClusterSize': config['cluster_size'],
+                'InstanceType': config['instance_type'],
+                'AmiCode': config['instance_ami']})})
+        return super().create(config)
+
+    def get_hosts(self, info):
+        return self.get_stack(info).get_host_ips()
+
+    def describe(self, info):
+        return {'host_list': launch.util.convert_host_list(self.get_hosts())}
+
+    def test(self, info, args, env):
+        raise NotImplementedError('Bare clusters cannot be tested!')
```

Comment:
That is not true. onprem clusters do support being tested, but this is actually the harness for the underlying homogeneous instance group without DC/OS installed so it cannot be tested.